File: workflow-service/src/model/petri-net/petri-net.ts
export type PlaceId = number;

export type TransitionId = string;

export enum TransitionType {
  START,
  END,
  TASK,
  XOR_SPLIT,
  XOR_JOIN,
  AND_SPLIT,
  AND_JOIN,
}

export interface Transition {
  id: TransitionId;
  type: TransitionType;
  name?: string;
  incomingPlaces: PlaceId[];
  outgoingPlaces: PlaceId[];
}

export type PetriNetId = string;

export interface PetriNet {
  id: PetriNetId;
  placeCount: number;
  startPlace: PlaceId;
  transitions: Transition[];
}

File: proof-service/workflow/petri_net.go
package workflow

type PlaceId uint

type TransitionId string

type Transition struct {
	Id             TransitionId
	IncomingPlaces []PlaceId
	OutgoingPlaces []PlaceId
}

type PetriNetId string

type PetriNet struct {
	Id          PetriNetId
	PlaceCount  uint
	StartPlace  PlaceId
	Transitions []Transition
}
